src/main/webapp/scripts/group-details-script.js

```diff
@@ -0,0 +1,299 @@
+            document.getElementById("view-topics-group-managers"),
+            document.getElementById("view-topics-group-members"),
+            document.getElementById("view-topics-entire-organization"),
+            document.getElementById("view-topics-external"),],
+        "ALL_IN_DOMAIN_CAN_VIEW": [document.getElementById("view-topics-group-owners"),
+            document.getElementById("view-topics-group-managers"),
+            document.getElementById("view-topics-group-members"),
+            document.getElementById("view-topics-entire-organization"),],
+        "ALL_MEMBERS_CAN_VIEW": [document.getElementById("view-topics-group-owners"),
+            document.getElementById("view-topics-group-managers"),
+            document.getElementById("view-topics-group-members"),],
+        "ALL_MANAGERS_CAN_VIEW": [document.getElementById("view-topics-group-owners"),
+            document.getElementById("view-topics-group-managers"),],
+        "ALL_OWNERS_CAN_VIEW": [document.getElementById("view-topics-group-owners"),],
+    }
+    // publish posts
+    var whoCanPostMessageCheckboxMap = {
+        "ANYONE_CAN_POST": [document.getElementById("publish-posts-group-owners"),
+            document.getElementById("publish-posts-group-managers"),
+            document.getElementById("publish-posts-group-members"),
+            document.getElementById("publish-posts-entire-organization"),
+            document.getElementById("publish-posts-external"),],
+        "ALL_IN_DOMAIN_CAN_POST": [document.getElementById("publish-posts-group-owners"),
+            document.getElementById("publish-posts-group-managers"),
+            document.getElementById("publish-posts-group-members"),
+            document.getElementById("publish-posts-entire-organization"),],
+        "ALL_MEMBERS_CAN_POST": [document.getElementById("publish-posts-group-owners"),
+            document.getElementById("publish-posts-group-managers"),
+            document.getElementById("publish-posts-group-members"),],
+        "ALL_MANAGERS_CAN_POST": [document.getElementById("publish-posts-group-owners"),
+            document.getElementById("publish-posts-group-managers"),],
+        "ALL_OWNERS_CAN_POST": [document.getElementById("publish-posts-group-owners"),],
+        "NONE_CAN_POST": [],
+    }
+    // manage members
+    var whoCanModifyMembersCheckboxMap = {
+        "ALL_MEMBERS": [document.getElementById("manage-members-group-owners"),
+            document.getElementById("manage-members-group-managers"),
+            document.getElementById("manage-members-group-members"),],
+        "OWNERS_AND_MANAGERS": [document.getElementById("manage-members-group-owners"),
+            document.getElementById("manage-members-group-managers"),],
+        "OWNERS_ONLY": [document.getElementById("manage-members-group-owners"),],
+        "NONE": [],
+    }
+    var settingsCheckboxes = [];
+    settingsCheckboxes.push(whoCanContactOwnerCheckboxMap[json.whoCanContactOwner]);
```

Comment:
rather than use SettingsCheckboxes and the for loop below, you can get all the values of each whoCan... variable with Object.values(json). Then you can pass them to a method which marks all of them as true